src/v/storage/segment_reader.cc

```diff
@@ -89,4 +208,47 @@ std::ostream& operator<<(std::ostream& os, const segment_reader_ptr& seg) {
     }
     return os << "{{log_segment: null}}";
 }
+
+segment_reader_handle::segment_reader_handle(segment_reader* parent)
```

Comment:
`segment_reader` is moveable, using raw pointer may be dangling when reader is moved.